src/modules/rc_update/rc_update.cpp

```diff
@@ -598,15 +599,28 @@ void RCUpdate::UpdateManualSwitches(const hrt_abstime &timestamp_sample)
 		// slots. And finally we add half a slot width to ensure that integer rounding
 		// will take us to the correct final index.
 		const float value = _rc.channels[_param_rc_map_fltmode.get() - 1];
-		switches.mode_slot = (((((value - slot_min) * num_slots) + slot_width_half) / (slot_max - slot_min)) +
-				      slot_width_half) + 1;
+		uint8_t switch_slot = (((((value - slot_min) * num_slots) + slot_width_half) / (slot_max - slot_min)) +
```

Comment:
@junwoo091400 The origins of this overly complicated formula are from when the mode switch feature was introduced:
https://github.com/PX4/PX4-Autopilot/pull/3919/files#diff-f0c485272a32f0acfd5f3719860d281d224f09bd7af63e85c82f0f9351ba8726R1907
https://github.com/PX4/PX4-Autopilot/pull/3998/files#diff-f0c485272a32f0acfd5f3719860d281d224f09bd7af63e85c82f0f9351ba8726R1921
We can simplify it but I intentionally didn't want to do that in this pr.